cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -84,7 +84,11 @@ type Etcd struct {
 type NodeConfiguration struct {
 	metav1.TypeMeta
 
-	Discovery Discovery
+	DiscoveryURL      string
+	DiscoveryFile     string
+	DiscoveryToken    string
+	TLSBootstrapToken string
+	Token             string
```

Comment:
Hmm -- we say "master" but it really isn't a thing.  APIServer is technically more correct.  But I won't bikeshed on it.